internal/pagination/pagination.go

```diff
@@ -124,36 +137,32 @@ func PaginateRequest[T any, PbT ResponseItem](
 	// request page size+1 so we can tell if we're at the end
 	limit := pageSize + 1
 
-	finalItems, completeListing, err := fillPage(ctx, limit, pageSize, refreshToken, itemLister, converterAndFilterer)
+	resp.Items, resp.CompleteListing, err = fillPage(ctx, limit, pageSize, refreshToken, itemLister, converterAndFilterer)
 	if err != nil {
 		return nil, errors.Wrap(ctx, err, op)
 	}
 
-	resp := &ListResponse[PbT]{
-		Items:           finalItems,
-		CompleteListing: completeListing,
-	}
-
-	if len(finalItems) > 0 {
+	// Only include a refresh token if we received a result
+	// or we have an incoming refresh token, so that we always have
+	// something to populate LastItemId and LastItemUpdatedTime with.
```

Comment:
I originally had this PR allow last item ID and last item updated time to be empty (equivalent to a sentinel value, really). It means that validation now can't validate the ID or the update time for one, and it also means that all pagination implementations have to know and care about whether a non-nil refresh token actually has a last item or not (this logic is not possible to make generic).

I don't think there are any cases where returning a refresh token for completely empty responses (with no existing refresh token) makes a difference compared to returning no refresh token in that case. We wouldn't use the last item ID in that refresh token, obviously, and we also wouldn't use the create time, since any items that may have been deleted won't be known to the caller anyway, or they would have been returned in the original request.

In the end, I thought guaranteeing that a refresh token always has a last item outweighed the benefit of returning a refresh token in all cases. I agree that there's a nice symmetry to always returning a refresh token but I don't think it's worth the added complexity.